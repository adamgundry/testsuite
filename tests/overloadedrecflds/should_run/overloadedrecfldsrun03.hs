{-# LANGUAGE OverloadedRecordFields, TypeFamilies #-}

data family F a

data instance F Int  = MkFInt  { foo :: Int }
data instance F Bool = MkFBool { bar :: Bool }


data family G a

data instance G Int = MkGInt { foo :: Int }
data instance G Bool = MkGBool { bar :: Bool }


main = do print (foo (MkFInt 42))
          print (foo (MkGInt 42))
          print (bar (MkFBool True))
          print (bar (MkGBool True))
