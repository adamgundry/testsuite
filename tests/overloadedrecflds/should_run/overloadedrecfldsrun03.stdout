42
42
True
True
